src/main/webapp/script.js

```diff
@@ -157,6 +158,66 @@ function fetchAuthIndexQuestion() {
   })
 }
 
+/**
+ * Displays logout button or redirects to index in verification page.
+ */
+function fetchAuthVerification() {
+  fetch('/authentication').then(response => response.json()).then(user => {
+    if (user.isUserLoggedIn) {
+      // If user is logged in, show logout button in navbar.
+      if (!user.isUserRegistered) {
+        // If logged in user is not registered, redirect to signup page.
+        window.location.replace('/signup.html');
+      }
+
+      // Add logout button to navbar.
+      createAuthenticationButton(
+          user.authenticationUrl, 'btn-outline-success', 'Log Out', 'login');
+    } else {
+      // If user is logged out, show signup and login buttons in navbar.
+      window.location.replace('/index.html');
+    }
+  })
+}
+
+/**
+ * Fetches the question list in the index and adds pagination elements.
+ * Used for homepage and searching.
+ * @param {int} pageNumber : current page number to display.
+ */
+async function fetchForum(pageNumber) {
+  // Question ID -1 tells the server to fetch all questions.
+  const response = await fetch('/fetch-questions?id=-1&page=' + pageNumber);
+  const questionsObject = await response.json();
+
+  const questionsContainer = document.getElementById('forum');
+
+  // Empty the HTML for multiple searches in a row.
+  questionsContainer.innerHTML = '';
+  questionsContainer.appendChild(createPageElement(
+      questionsObject, pageNumber, /**hasRedirect=*/true, /**isSearch=*/false));
+}
+
+/**
+ * Fetches a question's data for the single question view.
+ */
+async function fetchSingleQuestion() {
+  const questionId = (new URL(document.location)).searchParams.get("id");
+  const response = await fetch('/fetch-questions?id=' + questionId + '&page=-1');
+  const questionsObject = await response.json();
+  const questionsContainer = document.getElementById('question');
+  
+  if (questionsObject.length > 0) {
```

Comment:
Again, I'm a bit confused. This is supposed to fetch just one question, right? Why do you have to iterate through many of them?

Reply:
yep, I changed it